src/main/java/edu/harvard/iq/dataverse/DataFileServiceBean.java

```diff
@@ -1435,4 +1445,15 @@ public boolean isReplacementFile(DataFile df) {
         }
         return false;
     }  // end: isReplacementFile
+    
+    public List<Long> selectFilesWithMissingOriginalTypes() {
+        Query query = em.createNativeQuery("SELECT f.id FROM datafile f, datatable t where t.datafile_id = f.id AND t.originalfileformat='" + MIME_TYPE_TAB + "' ORDER BY f.id");
```

Comment:
what about `TypedQuery<Long> query = em.createQuery("SELECT f.id FROM datafile f, datatable t where t.datafile_id = f.id AND t.originalfileformat='" + MIME_TYPE_TAB + "' ORDER BY f.id", Long.class);`?